### PR TITLE
New Search bar design

### PR DESCRIPTION
**Pull request by spoorthyv**
Updated the search design. Its now bigger with placeholder text and a blue icon to draw attention to it. When we ran user tests we realized that basically no one was able to find the search bar unless they were explicitly looking for it (and even then it was hard). 

Im kinda unsure if the way I implemented it is acceptable.

screen shot 2016-07-21 at 11 05 51 am

### CODE REVIEW

File: src/filebrowser/browser.ts [hunk 1]
```diff
@@ -299,10 +299,17 @@ class FileBrowserWidget extends Widget {
...
+   protected onAfterAttach(msg: Message): void {
+     super.onAfterAttach(msg);
+     this.refresh();
+     let inputWrapper = document.body.querySelector('.p-CommandPalette-inputWrapper');
+     let searchIcon = document.createElement('i');
+     searchIcon.className = 'fa fa-search p-CommandPalette-searchIcon';
+     inputWrapper.appendChild(searchIcon);
+     let inputNode = document.body.querySelector('.p-CommandPalette-input') as HTMLInputElement;
+     inputNode.placeholder = "SEARCH";
+     console.log("yello");
```
Comment (jasongrout) [inline src/filebrowser/browser.ts]: delete this :)

File: src/filebrowser/browser.ts [hunk 2]
```diff
@@ -299,10 +299,17 @@ class FileBrowserWidget extends Widget {
...
    * A message handler invoked on an `'after-attach'` message.
    */
-  protected onAfterAttach(msg: Message): void {
-    super.onAfterAttach(msg);
-    this.refresh();
-  }
+   protected onAfterAttach(msg: Message): void {
+     super.onAfterAttach(msg);
+     this.refresh();
+     let inputWrapper = document.body.querySelector('.p-CommandPalette-inputWrapper');
```
Comment (jasongrout) [inline src/filebrowser/browser.ts]: Even better: `this.node.querySelector...` scopes it to just this node.

File: src/filebrowser/browser.ts [hunk 3]
```diff
@@ -299,10 +299,17 @@ class FileBrowserWidget extends Widget {
...
+     let searchIcon = document.createElement('i');
+     searchIcon.className = 'fa fa-search p-CommandPalette-searchIcon';
+     inputWrapper.appendChild(searchIcon);
+     let inputNode = document.body.querySelector('.p-CommandPalette-input') as HTMLInputElement;
```
Comment (jasongrout) [inline src/filebrowser/browser.ts]: `this.node.querySelector...`

**Comment by jasongrout** [comment]
Looks good to me, modulo some minor comments

**Comment by jasongrout** [comment]
Oh, I was getting confused. `this.node.querySelector` won't work because the search box is not a descendent of the file browser. We shouldn't put it in the onAttach for the file browser.

**Comment by jasongrout** [comment]
@blink1073, @sccolbert - is phosphide the one putting the command palette on the left? Perhaps phosphide should provide a command palette, but not put it anywhere by default, so that the main app can put the command palette on the left, right, center, or have some other way of showing it. That would help in this situation, and be more general overall.

**Comment by blink1073** [comment]
We are using an explicit extension to add it to the right (in our bootstrap file).

**Comment by blink1073** [comment]
https://github.com/jupyter/jupyterlab/blob/12048d6efb552cc850ee0e07c49ff6f35ee27e07/jupyterlab/index.js#L32

**Comment by jasongrout** [comment]
Ah, so we write our own (see  https://github.com/phosphorjs/phosphide/blob/master/src/extensions/commandpalette.ts) in jupyterlab, instead of using the phosphide one. Then @spoorthyv could add his logic there.

**Comment by blink1073** [comment]
Yep, sounds good.

**Comment by sccolbert** [comment]
Yeah - monorepo will make this all easier.

**Comment by spoorthyv** [comment]
Should I wait until the monorep is implemented?

**Comment by jasongrout** [comment]
@spoorthyv - that sounds like a good idea. It's not a super high priority to get in right away.

**Comment by afshin** [comment]
@spoorthyv This PR is no longer blocked now that there is a `placeholder` attribute in the `ICommandPalette` interface for JupyterLab.

Depending on how you want to handle it, you may decide to blow away this branch and just copy your work over into a new branch based on the current `master` instead of dealing with merge conflicts.

If you run into issues, we can help!

**Comment by spoorthyv** [comment]
Sounds good. Ill get to work on this.

**Comment by spoorthyv** [comment]
I was able to add the placeholder text but I cant figure out how to add new DOM elements. I tried creating a new class that extended CommandPalette and added a onAfterAttatch method in the new class but that didnt work.

**Comment by afshin** [comment]
@spoorthyv I'm actually going to be gone for the next week, so I'm CCing @jasongrout and @blink1073 

The actual DOM structure of the command palette is not meant to be altered because the underlying class is relying on that structure do do things. Do you have an image of the design you're looking to implement? Maybe it'll give some insight as to what sort of interface you actually need `ICommandPalette` to fulfill.

**Comment by spoorthyv** [comment]
Yeah. I put a screenshot of the design in the original post. I need to add a search icon to the right of the search bar.

**Comment by sccolbert** [comment]
You can probably do this entirely with CSS using an `::after` rule on `.p-CommandPalette-search`.

**Comment by spoorthyv** [comment]
I created a new PR for this because of some personal github issues. #673

**Comment by blink1073** [comment]
Closing in favor of #673, thanks!

**Pull request closed without merging**